src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'removeProperties') {
+      for (const el of action.keysToRemove) {
+        delete copyState[el];
+      }
+    }
+
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    }
+
+    if (action.type === 'clear') {
+      for (const key in copyState) {
+        delete copyState[key];
+      }
+    }
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others